pkg/kvstore/etcd_test.go

```diff
@@ -1564,3 +1565,57 @@ func (e *EtcdLockedSuite) TestListPrefixIfLocked(c *C) {
 		c.Assert(err, IsNil)
 	}
 }
+
+func TestGetSvcNamespace(t *testing.T) {
+	type args struct {
+		address string
+	}
+	tests := []struct {
+		name          string
+		args          args
+		wantSvcName   string
+		wantNamespace string
+		wantErr       bool
+	}{
+		{
+			name: "test-1",
+			args: args{
+				address: "http://foo.bar.something",
+			},
+			wantSvcName:   "foo",
+			wantNamespace: "bar",
+			wantErr:       false,
+		},
+		{
+			name: "test-2",
+			args: args{
+				address: "http://foo.bar",
+			},
+			wantSvcName:   "foo",
+			wantNamespace: "bar",
+			wantErr:       false,
+		},
+		{
+			name: "test-3",
+			args: args{
+				address: "http://foo",
```

Comment:
What do we accept for `[optional]` at the end? ports, paths, ...? I guess in the end we just use `url.Parse()` so not a big deal to test, but it would provide examples in our tree if we had unit tests.

Reply:
add unit tests